pkg/validations/upgradevalidations/immutable_fields.go

```diff
@@ -79,23 +79,24 @@ func ValidateImmutableFields(ctx context.Context, k validations.KubectlClient, c
 	oldAWSIamConfigRef := &v1alpha1.Ref{}
 
 	for _, oIdentityProvider := range oSpec.IdentityProviderRefs {
-		switch oIdentityProvider.Kind {
-		case v1alpha1.AWSIamConfigKind:
-			oIdentityProvider := oIdentityProvider // new variable scoped to the for loop Ref: https://github.com/golang/go/wiki/CommonMistakes#using-reference-to-loop-iterator-variable
+		if oIdentityProvider.Kind == v1alpha1.AWSIamConfigKind {
```

Comment:
`oIdentityProvider := oIdentityProvider` dont we still need this due to the github link mentioned? I remember this was an issue way back.

Reply:
Not required anymore, as we are breaking out of the loop immediately and not iterating further